### PR TITLE
generate opening balances REM-326

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
fi-gen-open-bal

### CODE REVIEW

File: src/pages/fi-gen-open-bal/form/GenerateOpening.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      tbendDate: yup.string().required(' '),
+      tbFiscalYear: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      try {
+        const { recordId, ...rest } = obj
```
Comment (RebeccaFrem02) [inline src/pages/fi-gen-open-bal/form/GenerateOpening.js:33]: you may need to use formatDateToApi() check it

File: src/pages/fi-gen-open-bal/form/GenerateOpening.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  const { formik } = useForm({
+    initialValues: { fiscalYear: null, tbFiscalYear: null, recordId: 'N/A', endDate: null },
+    enableReinitialize: true,
+    maxAccess: access,
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      tbendDate: yup.string().required(' '),
```
Comment (RebeccaFrem02) [inline src/pages/fi-gen-open-bal/form/GenerateOpening.js:28]: the date is not turning red when submitting it empty

File: src/pages/fi-gen-open-bal/form/GenerateOpening.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    onSubmit: async obj => {
+      try {
+        const { recordId, ...rest } = obj
+
+        const response = await postRequest({
+          extension: RGFinancialRepository.FiOpeningBalance.gen,
+          record: JSON.stringify({ ...obj, tbendDate: formatDateToApi(obj.tbendDate) })
+        })
+
+        toast.success('Record Success')
```
Comment (hadichahine) [inline src/pages/fi-gen-open-bal/form/GenerateOpening.js:41]: platform labels

File: src/pages/fi-gen-open-bal/form/GenerateOpening.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        invalidate()
```
Comment (hadichahine) [inline src/pages/fi-gen-open-bal/form/GenerateOpening.js:43]: where is this declared? I can't see no invalidate. This errors and stays hidden.

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**